inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -47,8 +69,25 @@ export class CanvasGraphService {
         this._height = canvas.height;
         this._ticks = [];
         this._panPosition = null;
+        this._hoverPosition = null;
         this._positions = new Map<string, number>();
         this._datasetBounds = new Map<string, IPerfIndexBounds>();
+        this._globalTimeMinMax = {min: Infinity, max: 0};
+        this._drawableArea = {top: 0, left: 0, right: 0, bottom: 0};
+
+        if (this._ctx) {
+            const defaultMetrics = this._ctx.measureText(alphabet);
+            this._defaultLineHeight = defaultMetrics.actualBoundingBoxAscent + defaultMetrics.actualBoundingBoxDescent;
+
+            this._ctx.save();
+            this._ctx.font = tooltipFont;
+            const fontMetrics = this._ctx.measureText(alphabet);
+            this._tooltipLineHeight = fontMetrics.actualBoundingBoxAscent + fontMetrics.actualBoundingBoxDescent;
+            this._ctx.restore();
+        } else {

```

Comment:
Should we not completely fail the display if no canvas/context are available ?

Reply:
Yeah I'm actually kind of confused on how we would do that, do you have any suggestions?

Reply:
May be throwing an error in the constructor ? or just returning ?

Reply:
I throw an error now!